ee/vellum_cli/tests/test_push.py

```diff
@@ -171,3 +171,75 @@ class ExampleWorkflow(BaseWorkflow):
 
     extracted_files = _extract_tar_gz(call_args["artifact"].read())
     assert extracted_files["workflow.py"] == workflow_py_file_content
+
+
+def test_push__check_option_returns_report(mock_module, vellum_client):
+    # GIVEN a single workflow configured
+    temp_dir = mock_module.temp_dir
+    module = mock_module.module
+
+    # AND a workflow exists in the module successfully
+    base_dir = os.path.join(temp_dir, *module.split("."))
+    os.makedirs(base_dir, exist_ok=True)
+    workflow_py_file_content = """\
+from typing import Dict
+from vellum.workflows import BaseWorkflow
+from vellum.workflows.nodes import BaseNode
+from vellum_ee.workflows.display.nodes import BaseNodeDisplay
+
+class NotSupportedNode(BaseNode):
+    pass
+
+class NotSupportedNodeDisplay(BaseNodeDisplay[NotSupportedNode]):
+    def serialize(self, display_context, **kwargs) -> Dict:
+        raise NotImplementedError(f"Serialization is not supported.")
+
+class ExampleWorkflow(BaseWorkflow):
+    graph = NotSupportedNode
+"""
+    with open(os.path.join(temp_dir, *module.split("."), "workflow.py"), "w") as f:
+        f.write(workflow_py_file_content)
+
+    # AND the push API call returns successfully
+    vellum_client.workflows.push.return_value = WorkflowPushResponse(
+        workflow_sandbox_id=str(uuid4()),
+    )
+
+    # WHEN calling `vellum push`
+    runner = CliRunner(mix_stderr=True)
+    result = runner.invoke(cli_main, ["push", module, "--check"])
+
+    # THEN it should succeed
+    assert result.exit_code == 0
+
+    # AND we should have called the push API with the check option
+    vellum_client.workflows.push.assert_called_once()
+    call_args = vellum_client.workflows.push.call_args.kwargs
+    assert call_args["request_options"]["additional_body_parameters"]["check"] is True
+
+    # AND the report should be in the output
+    assert (
+        result.output
+        == """\x1b[38;20mLoading workflow from examples.mock\x1b[0m
+\x1b[38;20mRan check on Mock\x1b[0m
+"""
+    )
+
+
+# TODO: Implement in Vellum push `check` and output results
+# https://app.shortcut.com/vellum/story/5416
```

Comment:
This is what we would log to the CLI upon running `--check`. Alternatively, we could write to a `.log` file and output the path to that file here